### PR TITLE
Create no-multiple-api-calls rule

### PR DESCRIPTION
**Pull request by luacmartins**
- Implements `no-multiple-api-calls` rule.
- Adds `deprecatedAPI` to `no-api-in-views` and update tests.
- Fixes lint errors in other files.

# Issue
Fixes https://github.com/Expensify/Expensify/issues/211243

# Tests
1. Ran `npm run test` and make sure all tests pass.

### CODE REVIEW

File: eslint-plugin-expensify/no-multiple-api-calls.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const _ = require('underscore');
+const {isInActionFile} = require('./utils');
+const message = require('./CONST').MESSAGE.NO_MULTIPLE_API_CALLS;
+
+module.exports = {
+    create(context) {
+        function checkFunctionBody(node) {
+            // This rule is scoped to the ./actions/ directory because we already have a rule
```
Comment (roryabraham) [inline eslint-plugin-expensify/no-multiple-api-calls.js:8]: I don't think we need to scope this rule to the `./actions/` directory – we can have a rule preventing any API calls outside of actions and a separate rule for preventing multiple API calls in a single function. Those don't conflict with eachother, so let's keep it simple/more flexible and get rid of the `isInActionFile` check here.

File: eslint-plugin-expensify/CONST.js [hunk 2]
```diff
@@ -11,5 +11,6 @@ module.exports = {
         PREFER_ONYX_CONNECT_IN_LIBS: 'Only call Onyx.connect() from inside a /src/libs/** file. React components and non-library code should not use Onyx.connect()',
         PREFER_UNDERSCORE_METHOD: 'Prefer \'_.{{method}}\' over the native function.',
         PREFER_STR_METHOD: 'Prefer \'Str.{{method}}\' over the native function.',
+        NO_MULTIPLE_API_CALLS: 'Do not call API or deprecatedAPI multiple times in the same method. The API response should return all the necessary data in a single request.',
```
Comment (roryabraham) [inline eslint-plugin-expensify/CONST.js:14]:
```suggestion
        NO_MULTIPLE_API_CALLS: 'Do not call API or deprecatedAPI multiple times in the same method. The API response should return all the necessary data in a single request, and API calls should not be chained together.',
```

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: eslint-plugin-expensify/no-multiple-api-calls.js (hunk 1 above)
Comment (luacmartins) [reply eslint-plugin-expensify/no-multiple-api-calls.js:8]: Sounds good!

**Comment by luacmartins** [comment]
Updated!

**Review by roryabraham — DISMISSED** [review]
LGTM, all yours @marcaaron

File: eslint-plugin-expensify/tests/no-multiple-api-calls.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    deprecatedAPI.call(params);
+                }
+            `,
+        },
+    ],
+    invalid: [
+        {
+            code: `
+                function test() {
+                    deprecatedAPI.call(params).then(res => API.call(params2));
```
Comment (marcaaron) [inline eslint-plugin-expensify/tests/no-multiple-api-calls.test.js:33]:
```suggestion
                    DeprecatedAPI.CreateLogin(params).then(res => API.write('SomeCommand', params2));
```

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: eslint-plugin-expensify/tests/no-multiple-api-calls.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        sourceType: 'module',
+    },
+});
+
+ruleTester.run('no-multiple-api-calls', rule, {
+    valid: [
+        {
+            code: `
+                function test() {
+                    API.call(params);
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply eslint-plugin-expensify/tests/no-multiple-api-calls.test.js:17]: Makes sense! I'll update all of them. As always, thanks for the feedback!

**Comment by luacmartins** [comment]
Updated!

**Review by marcaaron — APPROVED** [review]

**Pull request merged**